src/udiskslinuxencrypted.c

```diff
@@ -408,8 +459,24 @@ handle_unlock (UDisksEncrypted        *encrypted,
   /* calculate the name to use */
   if (is_in_crypttab && crypttab_name != NULL)
     name = g_strdup (crypttab_name);
+  else {
+    if (is_luks)
+      name = g_strdup_printf ("luks-%s", udisks_block_get_id_uuid (block));
+    else
+      /* TCRYPT devices don't have a UUID, so we use the device number instead */
+      name = g_strdup_printf ("tcrypt-%" G_GUINT64_FORMAT, udisks_block_get_device_number (block));
+  }
+
+  /* save old encryption type to be able to restore it */
+  old_hint_encryption_type = udisks_encrypted_dup_hint_encryption_type(encrypted);
+
+  /* Set hint_encryption type. We have to do this before the
+   * actual unlock, in order to have this set before the device
+   * update triggered by the unlock. */
+  if (is_luks)
+    {} /* LUKS is detectable, so we don't have to set the hint */
```

Comment:
Please squash the comments and do ``if (!is_luks)``

Reply:
Done in ec71668d8b9c3beef3767d40382cb5fcc722c9ac.